### PR TITLE
Fix and update tests

### PR DESCRIPTION
**Pull request by t0mhaines**
Aiming for 85%+ Coverage

### CODE REVIEW

**Comment by t0mhaines** [comment]
This is as far as we can get as far as unit testing the hub. Coverage should go up once we have implemented #13

File: .travis.yml [hunk 1]
```diff
@@ -22,3 +22,4 @@ after_success:
 env:
   global:
   - MIX_ENV=test
+  - CIRCUITS_MIX_ENV=test
```
Comment (nelsonic) [inline .travis.yml:26]: @th0mas would be good to get this environment variable documented somewhere.
e.g. an in-line comment:
```suggestion
  # GPIO testing, see: https://github.com/elixir-circuits/circuits_gpio#testing
  - CIRCUITS_MIX_ENV=test
```

File: config/config.exs [hunk 2]
```diff
@@ -20,7 +20,9 @@ config :nerves, source_date_epoch: "1594028364"
 # Use Ringlogger as the logger backend and remove :console.
 # See https://hexdocs.pm/ring_logger/readme.html for more information on
 # configuring ring_logger.
-
+if Mix.env() == :test do
+  config :logger, level: :error
```
Comment (nelsonic) [inline config/config.exs:24]: 👍

**Review by nelsonic — APPROVED** [review]
@th0mas great additions for tests. Thanks! 🥇

**Comment by nelsonic** [comment]
Still getting _intermittent_ build (test) failure: 
https://travis-ci.com/github/dwyl/smart-home-firmware/builds/178730311#L561
image

Going to _restart_ the build and see if it passes. 🤞 
This should be addressed as part of #13

**Comment by nelsonic** [comment]
After re-running the build _two_ tests fail instead of one ... lol.
image

This definitely needs to be addressed. 

All tests pass on the 4<sup>th</sup> run on travis:

https://travis-ci.com/github/dwyl/smart-home-firmware/builds/178730311#L538
image

@th0mas ideally we wouldn't have any warnings either. 💭

**Comment by t0mhaines** [comment]
4 of those warnings are my fault, however two are due to a wrong behaviour specification in the PN532 library. I'll file an issue in that repo

**Comment by nelsonic** [comment]
@th0mas yeah, it's always good to have a trail of _why_ things are the way they are in the issues. 👍 
Otherwise great work, keep it up! 🙌

**Pull request merged**